components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementProviderServiceImpl.java

```diff
@@ -1113,21 +1461,39 @@ public PolicyMonitoringManager getPolicyMonitoringManager(String deviceType) {
 
     @Override
     public List<Device> getDevicesOfUser(String username, String deviceType) throws DeviceManagementException {
-        return  this.getDevicesOfUser(username, deviceType, true);
+        return this.getDevicesOfUser(username, deviceType, true);
     }
 
     @Override
     public List<Device> getDevicesOfUser(String username, String deviceType, boolean requireDeviceInfo) throws
             DeviceManagementException {
+        if (username != null && deviceType != null) {
```

Comment:
Check for empty strings as well. Use builtin methods like StringUtils.isBlank or appropriate. Please do the same for all applicable places.